design-proposals/dynamic-primary-pod-nic-name.md

```diff
@@ -0,0 +1,481 @@
+- Support Istio integration with Masquerade binding and a custom primary interface name.
+
+## Definition of Users
+
+- VM owner
+
+## User Stories
+
+As a VM owner:
+- I want my VMs to be able to bind the interface destined to pod network, to the pod's primary interface, no matter its name.
+- I want my VMs to be able to connect to the cluster's default network, as they do today.
+- I want my VMs' interface status reporting to be accurate when the interface destined to pod network is bound to a pod interface with a custom name.
+- I want to be able to use core bindings and network binding plugins when the primary pod interface has a custom name.
+- I want to be able to migrate my VM when the pod's primary interface has a custom name.
+
+## Repos
+
+- https://github.com/kubevirt/kubevirt
+
+# Design
+
+## Assumptions
+
+The KubeVirt components participating in this proposal are:
+- VMI controller
+- virt-handler
+- virt-launcher
+
+The following table describes which information is accessible to each relevant KubeVirt component:
+
+| Component/has access to | VirtualMachineInstance | Pod   | Cluster-config |
+|-------------------------|------------------------|-------|----------------|
+| VMI controller          | True                   | True  | True           |
+| virt-handler            | True                   | False | True           |
+| virt-launcher           | True                   | False | False          |
+
+Cluster wide config passes from virt-handler to virt-launcher via their dedicated gRPC channel.
+
+The pod interfaces' naming scheme is currently independently calculated at each of the three components, based on input from VMI.Spec.
+
+The information about the pod's primary interface name should be propagated to all three components. 
+
+## Preferred alternative
+The preferred alternative is inferring the primary interface name from Multus’ network-status annotation and reporting the pod interface name per interface.
+
+## Inferring the primary interface name from Multus’ network-status annotation
+
+When KubeVirt is used in a cluster that has Multus as its default CNI, Multus adds the
+`k8s.v1.cni.cncf.io/network-status` annotation to the virt-launcher pod following its network attachment process.
+This annotation includes information about all the network interfaces attached to the pod.
+
+The VMI controller currently uses this annotation for:
+
+- Updating the VMI status interfaces section with the `multus-status` [info-source](https://kubevirt.io/api-reference/v1.3.1/definitions.html#_v1_virtualmachineinstancenetworkinterface).
+- Creating the `kubevirt.io/network-info` annotation, which is the data source for DownwardAPI when using SR-IOV NICs and certain network binding plugins.
+
+The `k8s.v1.cni.cncf.io/network-status` annotation shall be parsed to infer the default (primary) network interface instead of relying on a hard coded
```

Comment:
nit: "shall" feels a bit out of place, consider changing to "The annotation is parsed to infer...".